src/registry.rs

```diff
@@ -219,30 +219,37 @@ impl Registry {
     }
 }
 
+macro_rules! conditional_compile {
```

Comment:
I prefer writing register_process function for tow features instead of the macro.